grpc/resolver.go

```diff
@@ -0,0 +1,105 @@
+package grpc
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+
+	"google.golang.org/grpc/resolver"
+)
+
+// init registers the `staticBuilder` with the gRPC resolver registry.
+func init() {
+	resolver.Register(NewStaticBuilder())
+}
+
+// NewStaticBuilder creates a `staticBuilder` used to construct static DNS
+// resolvers.
+func NewStaticBuilder() resolver.Builder {
+	return &staticBuilder{}
+}
+
+// staticBuilder implements the `resolver.Builder` interface.
+type staticBuilder struct{}
+
+// Build constructs a `staticResolver` which implements the `resolver.Resolver`
+// interface. This method is typically called by the gRPC dialer, which passes a
+// list of comma separated IPv4/6 addresses and a `resolver.ClientConn`. The
+// state of the `resolver.ClientConn` is updated with the list of addresses and
+// the `resolver.ClientConn` is wrapped in the returned `staticResolver`.
```

Comment:
This half of this doccomment feels like it belongs on NewStaticResolver, rather than on Build, since it's describing the logic of that function.

Reply:
I'll go ahead and split these two up, thanks.